common/persistence/cassandra/errors.go

```diff
@@ -121,6 +130,13 @@ func extractErrors(
 
 	var errors []error
 
+	rowType, ok := conflictRecord["type"].(*int)
+	if !ok || rowType == nil {
+		return errors
+	}
+	// Dereference rowType for later use
+	conflictRecord["type"] = *rowType
+
```

Comment:
I finally understood what you try to achieve here. I don't think that you need `newConflictRecord` func. It should work with default `map[string]interface{}` and here code should be:
```suggestion
    // ScyllaDB will return rows with null values to match # of queries in a batch query (see #2683).
    // Field types will be pointer types (i.e. *int instead of int) in this case to support nil values.
    // Check only "type" field for simplicity.
	if rowType, ok := conflictRecord["type"].(*int); ok && rowType == nil {
		return errors
	}
```

Reply:
If I don't use `newConflictRecord`, then `ok` will be false when I try to get `conflictRecord["type"].(*int)` -- I'm not familiar enough with Go to know exactly why it works like that, but I'm assuming this is related to how the Cassandra client library works.

Reply:
Ok, there is definitely some sort of reflection magic inside Cassandra library. Can you try:
```go
fmt.Printf("%T\n, conflictRecord["type"])
```
w/o `newConflictRecord`. It should print underlying type of `"type"` key. What is it?

Reply:
It would be also great if you add test that fails in ScyllaDB w/o your fix and pass with it. Then I will be able to test it myself.

Reply:
> Ok, there is definitely some sort of reflection magic inside Cassandra library. Can you try:
> 
> ```go
> fmt.Printf("%T\n", conflictRecord["type"])
> ```
> 
> w/o `newConflictRecord`. It should print underlying type of `"type"` key. What is it?

If I am not mistaken, this part does Unmarshalling.
https://github.com/gocql/gocql/blob/7a6cf00bbc98f4d7037e4a0fcca96fc946fd63d6/marshal.go#L205

We also used `newConflictRecord` way in our project to solve this issue, but @PenguinToast was faster :D
It is working on live env and seems to be no issues.